pages/news-and-events/index.vue

```diff
@@ -103,7 +111,8 @@ export default Vue.extend<Data, never, Computed, never>({
       upcomingEvents: [],
       pastEvents: [],
       isShowingAllUpcomingEvents: false,
-      news: []
+      news: [],
+      heroData: {} as HeroDataEntry
```

Comment:
@matthewtgilbride I had to do this in order for TS to compile. This doesn't seem correct to me, but otherwise it complains about it now having the correct fields, or that it is not assignable to `Data`. What is the best practice here?

Reply:
@cameronbaney I actually think what you did is the best option.  Another option would be to make the type of heroData `HeroDataEntry | {}`, meaning "this thing is either a HeroDataEntry or it's the empty object".

The problem there is, every time you try to access it as a regular `HeroDataEntry`, TS is going to complain that any of the actual properties _might_ not be there.  That is objectively the more annoying alternative.  

There is probably a way shim the types such that the `data` key only has to return everything that is _not_ a part of `asyncData`, and `asyncData` only. has to return everything that is.  For now however I think your instincts are right.

Reply:
That makes sense. Thank you very much @matthewtgilbride